pyaedt/modeler/PrimitivesNexxim.py

```diff
@@ -1117,6 +1117,155 @@ def add_subcircuit_hfss_link(
+        """Add a subcircuit 2D Extractor link.
+
+        Parameters
+        ----------
+        pyaedt_app : :class:`pyaedt.q3d.Q3d` or :class:`pyaedt.q3d.Q2d` or :class:`pyaedt.q3d.Hfss`
+            pyaedt application object to include. It could be an Hfss object, a Q3d object or a Q2d.
+        solution_name : str, optional
+            Name of the solution and sweep. The
+            default is ``"Setup1 : Sweep"``.
+        extrusion_length : float, str, optional
+            Extrusion length for 2D Models. Default is 10 (in model units).
+        map_source_variables : bool, optional
+            Either if the dynamic link object has to map or not the source design variables.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Object3d.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+
+        >>> oModelManager.Add
+        >>> oComponentManager.Add
+        >>> oDesign.AddCompInstance
+        """
+        comp_name = generate_unique_name(pyaedt_app.design_name)
+        source_project_path = pyaedt_app.project_file
+        source_design_name = pyaedt_app.design_name
+        matrix = None
+        if pyaedt_app.design_type == "HFSS":
+            pin_names = pyaedt_app.get_excitations_name()
+        elif pyaedt_app.design_type == "Q3D Extractor":
+            excts = list(pyaedt_app.oboundary.GetExcitations())
+            i = 0
+            sources = []
+            sinks = []
+            while i < len(excts):
+                if excts[i + 1] == "Source":
+                    sources.append(excts[i])
+                elif excts[i + 1] == "Sink":
+                    sinks.append(excts[i])
+                i += 2
+            pin_names = sources + sinks
+        elif pyaedt_app.design_type == "2D Extractor":
+            excts = list(pyaedt_app.oboundary.GetExcitations())
+            pins = []
+            i = 0
+            while i < len(excts):
+                if excts[i + 1] != "ReferenceGround":
+                    pins.append(excts[i])
+                i += 2
+            pin_names = [i + "_in" for i in pins]
+            pin_names.append("Input_ref")
+            pin_names.extend([i + "_out" for i in pins])
+            pin_names.append("Output_ref")
+        variables = {}
+        if map_source_variables:
+            for k, v in pyaedt_app.variable_manager.variables.items():
+                variables[k] = v.string_value
+        if not solution_name:
+            solution_name = pyaedt_app.nominal_sweep
+        return self._add_subcircuit_link(
+            comp_name=comp_name,
+            pin_names=pin_names,
+            source_project_path=source_project_path,
+            source_design_name=source_design_name,
+            solution_name=solution_name,
+            image_subcircuit_path="",
+            model_type=pyaedt_app.design_type,
+            variables=variables,
+            extrusion_length_q2d=extrusion_length,
+            matrix=matrix,
```

Comment:
@maxcapodi78 I might be wrong but is seems that matrix will always be `None`, as it is set to `None` line 1210 and never changed after that. Is it expected?

Reply:
@maxcapodi78 , I agree with @MaxJPRey, matrix should be obtained before the call of _add_subcircuit_link(...), I think with the new PR (#834) this should be accessible